static/monitoring.js

```diff
@@ -109,6 +112,69 @@ function appendLine(element, text) {
     element.appendChild(document.createElement("br"));
 }
 
+function fetchHistoryNextPage() {
+    const next_page = document.getElementsByClassName("page_history_next");
```

Comment:
This doesn't seem like it will choose the right element, and `getElementsByClassName` also returns an array.

Reply:
You should probably define these as closures inside the function so they can capture the necessary state.